### PR TITLE
Fix run with Runme in Windows

### PR DESCRIPTION
**Pull request by christian-bromann**
The problem for Windows was not the storage path but more the way I passed along urls and Uris.

### CODE REVIEW

**Comment by sourishkrout** [comment]
@mxsdev could you do a review please? You were just deep into x-platform URIs etc.

I can do a final test on Windows once you're done.

File: src/extension/handler/utils.ts [hunk 1]
```diff
@@ -53,108 +54,107 @@ export async function getProjectDir () {
...
+  cb: (success: boolean) => void
+) {
+  const t = setTimeout(() => {
+    clearInterval(i)
+    window.showErrorMessage(`Timed out cloning repository ${repository}`)
+    return cb(false)
+  }, Math.max(config.get('timeout') || 0, MINIMAL_TIMEOUT))
+  const i = setInterval(async () => {
+    const dirEntries = await workspace.fs.readDirectory(targetDirUri).then((res) => res, (err) => {
+      console.error(`[Runme] Failed to fetch directory ${targetDirUri.toString()}: ${err.message}`)
```
Comment (mxsdev) [inline src/extension/handler/utils.ts:98]:
```suggestion
      console.error(`[Runme] Failed to fetch directory ${targetDirUri.fsPath}: ${err.message}`)
```

**Review by mxsdev — COMMENTED** [review]
Changes look good, though if you could clarify exactly what was going wrong with windows that would be great, since I can't say I totally understand where exactly that was fixed. I mean my assumption is that it was caused by this line:

```ts
const targetDirUri = Uri.joinPath(projectDirUri, await getTargetDirName(projectDirUri, suggestedProjectName))
```

Which you fixed.

Only other thing is that you can theoretically build stronger tests - see `tests/extension/configuration.test.ts`, which uses a more robust Uri mock. It also uses separate `path` modules (`path.posix` vs `path.win32`) to test path handling on different operating systems.

Feel free to do this or not, I just wanted to bring it up to avoid regressions down the line. Especially since (it looks like) all the tests in `uri.test.ts` and `utils.test.ts` are POSIX paths, so there's no coverage for the windows case in there.

File: src/extension/handler/utils.ts [hunk 2]
```diff
@@ -18,32 +19,32 @@ const MINIMAL_TIMEOUT = 10 * 1000
...
+    return context.globalStorageUri
+  }
 
-    const projectDir = Uri.parse(projectDirPath)
-    const isExisting = await workspace.fs.stat(projectDir)
-        .then(() => true, () => false)
-    if (isExisting) {
-        return projectDir
-    }
+  const projectDir = Uri.parse(url.pathToFileURL(projectDirPath).toString())
```
[the remark this answers is not in the document]
Comment (christian-bromann) [reply src/extension/handler/utils.ts:29]: ah neat

**Comment by sourishkrout** [comment]
@mxsdev @christian-bromann is this ready for a final functional test? I'd like to do one on my Windows VM. Please let me know.

**Comment by christian-bromann** [comment]
Just rebased after changes on the origin branch, running some tests myself.

**Comment by christian-bromann** [comment]
Ok, double checked on Windows and it works as expected. I also reverted the indentation change to make this easier to review. I will reapply it before merge.

**Comment by sourishkrout** [comment]
There's something odd going on, perhaps force pushes on a branch that depends on a branch, where below's line reintroduces the enclosing `Uri.parse` and the `.fsPath` in `Uri.parse(targetDirUri.fsPath)` which are redundant and break on Windows. I merged the upstream PR which changed the base to `main`. I would have expected a conflict on that line but didn't see one. Weirdly enough in the changeset on GitHub it's right but locally checked out it's wrong 🤷. Anyways, let's resolve the conflict and make sure the we're only passing `targetDirUri` which is already a valid vscode.Uri.

```
        await commands.executeCommand('vscode.openFolder', Uri.parse(targetDirUri.fsPath), {
            forceNewWindow: true
        })
```

**Review by sourishkrout — APPROVED** [review]
If version conflict is resolved properly - it should work 👍 . Please see https://github.com/stateful/vscode-runme/pull/264#issuecomment-1458300522.

**Pull request merged**